analysis/quantifier-comparisons/02-compare_quants.Rmd

```diff
@@ -0,0 +1,350 @@
+---
+title: "SCRNAseq quantification comparisons"
+author: "Joshua Shapiro for CCDL"
+output: 
+  html_notebook:
+    toc: true
+    toc_float: true
+---
+
+
+## Setup
+
+### Load Libraries
+```{r setup}
+library(ggplot2)
+library(magrittr)
+library(SingleCellExperiment)
+
+# set seed
+set.seed(2020)
+```
+
+
+
+### File and directory setup
+```{r}
+# Files and dirs generated by 01-import-quant-data.Rmd
+
+import_dir <- 'data'
+
+quant_info_file <- file.path(import_dir, 'quant_info.tsv')
+
+alevin_rds <- file.path(import_dir, 'alevin_sces.rds')
+kallisto_rds <- file.path(import_dir, 'kallisto_sces.rds')
+cellranger_rds <- file.path(import_dir, 'cellranger_sces.rds')
+
+
+```
+
+### Read in data files
+
+```{r}
+quant_info <- readr::read_tsv(quant_info_file )
+
+alevin_sces <- readr::read_rds(alevin_rds)
+kallisto_sces <- readr::read_rds(kallisto_rds)
+cellranger_sces <- readr::read_rds(cellranger_rds)
+```
+
+
+### Get Annotations from AnnotationHub
+
+```{r }
+hub = AnnotationHub::AnnotationHub(ask = FALSE)
+# Ensembl v100 Homo sapiens is AH79689
+ensdb = hub[["AH79689"]]
+ensg <- genes(ensdb)
+```
+
+Create vectors of mitochondrial genes and coding genes for later.
+
+```{r}
+# create the mitochondrial gene list
+mito_genes <- ensg[seqnames(ensg) == 'MT']$gene_id
+
+coding_genes <- ensg[ensg$gene_biotype == "protein_coding"]$gene_id
+```
+
+## Calculate cell QC statistics
+
+A small function to set default QC calculations for SCEs in a list:
+```{r}
+addPerCellQC_mito <- function(sce, mito = mito_genes){
+  scater::addPerCellQC(
+    sce, 
+    subsets = list(mito = mito[mito %in% rownames(sce)])
+  )
+}
+```
+
+Calculate cell QC statistics for each set of SCES
+
+```{r}
+alevin_sces <- alevin_sces %>% 
+  purrr::map(addPerCellQC_mito)
+
+kallisto_sces <- kallisto_sces %>% 
+  purrr::map(addPerCellQC_mito)
+
+cellranger_sces <- cellranger_sces %>% 
+  purrr::map(addPerCellQC_mito)
+```
+
+Make a data frames of the cell QCs.
+```{r}
```

Comment:
```suggestion

```{r}
```

Reply:
I think I got all of them, but either github is confused or I am... you don't really want 2 blank lines here do you?

Reply:
Nope, just one.